test/integration/dialects/postgres/query-interface.test.js

```diff
@@ -8,13 +8,151 @@ var chai = require('chai')
+          });
+        });
+        
+      });
+
+      it('produces an error when missing expected parameters', function () {
+        var self = this;
+        var body = 'return 1;';
+        var options = {};
+        
+        //requires functionName  
+        expect( function () {
+          self.queryInterface.createFunction(null, [{name:'test'}], 'integer', 'plpgsql', body, options);
+        }).to.throw(/createFunction missing some parameters. Did you pass functionName, returnType, language and body/);
+      
+        //requires Parameters array
+        expect( function () {
+          self.queryInterface.createFunction('create_job',null, 'integer', 'plpgsql', body, options);
+        }).to.throw(/function parameters array required/);
+        
+        //requires returnType
+        expect( function () {
+          self.queryInterface.createFunction('create_job', [{type:'varchar',name:'test'}], null, 'plpgsql', body, options);
+        }).to.throw(/createFunction missing some parameters. Did you pass functionName, returnType, language and body/);
+      
+        //requires type in parameter array
+        expect( function () {
+          self.queryInterface.createFunction('create_job', [{name:'test'}], 'integer', 'plpgsql', body, options);
+        }).to.throw(/function or trigger used with a parameter without any type/);
+        
+        //requires language
+        expect( function () {
+          self.queryInterface.createFunction('create_job', [{type:'varchar',name:'test'}], 'varchar', null, body, options);
+        }).to.throw(/createFunction missing some parameters. Did you pass functionName, returnType, language and body/);
+
+        //requires body
+        expect( function () {
+          self.queryInterface.createFunction('create_job', [{type:'varchar',name:'test'}], 'varchar', 'plpgsql', null, options);
+        }).to.throw(/createFunction missing some parameters. Did you pass functionName, returnType, language and body/);
+
+      });
+    });
+
+    describe('dropFunction',function () {
+      beforeEach( function () {
+
+        var body = 'return test;';
+        var options = {};
+
+        //make sure we have a droptest function in place.
+        return this.queryInterface.createFunction('droptest', [{type:'varchar',name:'test'}], 'varchar', 'plpgsql', body, options)
+        //suppress errors.. this could fail if the function is already there.. thats ok. 
+        .catch( function (err) { });
+      });
+
+      it('can drop a function', function () {
+        var self = this;
+         
+        //call dropFunction
+        return self.queryInterface.dropFunction('droptest',[{type:'varchar',name:'test'}])
+        //validate
+        .then( function () {
+          return self.sequelize.query('select droptest(\'test\');', { type: self.sequelize.QueryTypes.SELECT })
+          .then( function () {
+            throw new Error('function failed to drop');
```

Comment:
If droptest is not dropped as expected then  .then()  would be called because the droptest function will still exist.  The test will then produce an error stating the function failed to drop.